pydis_site/apps/api/models/bot/filters.py

```diff
@@ -0,0 +1,259 @@
+    )
+    infraction_duration = models.DurationField(
+        null=True,
+        help_text="The duration of the infraction. 0 for permanent."
+    )
+    infraction_channel = models.BigIntegerField(
+        validators=(
+            MinValueValidator(
+                limit_value=0,
+                message="Channel IDs cannot be negative."
+            ),
+        ),
+        help_text="Channel in which to send the infraction.",
+        null=True
+    )
+    guild_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers.",
+        null=True
+    )
+    filter_dm = models.BooleanField(help_text="Whether DMs should be filtered.", null=True)
+    dm_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers on a DM.",
+        null=True
+    )
+    remove_context = models.BooleanField(
+        help_text=(
+            "Whether this filter should remove the context (such as a message) "
+            "triggering it."
+        ),
+        null=True
+    )
+    bypass_roles = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Roles and users who can bypass this filter.",
+        null=True
+    )
+    enabled = models.BooleanField(
+        help_text="Whether this filter is currently enabled.",
+        null=True
+    )
+    send_alert = models.BooleanField(
+        help_text="Whether an alert should be sent.",
+        null=True
+    )
+
+    enabled_channels = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Channels in which to run the filter even if it's disabled in the category.",
+        null=True
+    )
+    disabled_channels = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Channels in which to not run the filter.", null=True
+    )
+    enabled_categories = ArrayField(
+        models.CharField(max_length=100),
+        help_text="The only categories in which to run the filter.",
+        null=True
+    )
+    disabled_categories = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Categories in which to not run the filter.",
+        null=True
+    )
+
+    def __str__(self) -> str:
+        return f"Filter {self.content!r}"
+
+    class Meta:
+        """Metaclass for FilterBase to make it abstract model."""
+
+        abstract = True
+
+
+class Filter(FilterBase):
+    """
+    The main Filter models based on `FilterBase`.
+
+    The purpose to have this model is to have access to the Fields of the Filter model
+    and set the unique constraint based on those fields.
+    """
+
+    class Meta:
+        """Metaclass Filter to set the unique constraint."""
+
+        constraints = (
+            UniqueConstraint(
+                fields=tuple(
+                    [field.name for field in FilterBase._meta.fields
+                     if field.name not in ("id", "description", "created_at", "updated_at")]
```

Comment:
Nitpick, you don't need the `[]` here and can avoid building the intermediate list :)
```suggestion
                fields=tuple(
                    field.name for field in FilterBase._meta.fields
                     if field.name not in ("id", "description", "created_at", "updated_at")
```

Reply:
[Use consistent quoting style](https://github.com/python-discord/site/pull/861/commits/91b89475913210400cf39884efe37ab5552efbf7)